ios/RNGeth.swift

```diff
@@ -82,16 +109,201 @@ class RNGeth: RCTEventEmitter, GethNewHeadHandlerProtocol {
+            reject(nil, nil, NSErr)
+        }
+    }
+
+    /**
+     * Unlock an account with a passphrase
+     *
+     * @param account String account to unlock
+     * @param passphrase String passphrase for the account
+     * @param timeout Int64 duration of unlock period
+     * @param promise Promise
+     * @return Return Boolean the unlock status
+     */
+    @objc(unlockAccount:passphrase:timeout:resolver:rejecter:)
+    func unlockAccount(account: NSString, passphrase: NSString, timeout: NSInteger, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
+        do {
+            guard let keyStore = geth_node.getKeyStore() else {
+                throw RuntimeError("KeyStore not ready")
+            }
+            let account = try geth_node.findAccount(rawAddress: account as String)!
+            let _ = try keyStore.timedUnlock(account, passphrase: passphrase as String, timeout: Int64(timeout))
+            resolve([true] as NSObject)
+        } catch let NSErr as NSError {
+            NSLog("@", NSErr)
+            reject(nil, nil, NSErr)
+        }
+    }
+
+    /**
+     * Add a new account to the keystore
+     *
+     * @param privateKey Data the private key to add
+     * @param passphrase String the passphrase to lock it with
+     * @param promise Promise
+     * @return Account the new account
+     */
+    @objc(addAccount:passphrase:resolver:rejecter:)
+    func addAccount(privateKeyHex: NSString, passphrase: NSString, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
+        do {
+            guard let keyStore = geth_node.getKeyStore() else {
+                throw RuntimeError("KeyStore not ready")
+            }
+            let privateKey = Data(hexString: privateKeyHex as String)
+            let account = try keyStore.importECDSAKey(privateKey, passphrase: passphrase as String)
+            resolve([account.getAddress()?.getHex()] as NSObject)
+        } catch let NSErr as NSError {
+            NSLog("@", NSErr)
+            reject(nil, nil, NSErr)
+        }
+    }
+
+    /**
+     * Signs data using an unlocked account
+     *
+     * @param data String hex encoded input
+     * @param signer String signer address
+     * @param promise Promise
+     * @return Return signed transaction
+     */
+    @objc(signHash:signer:resolver:rejecter:)
+    func signHash(data: NSString, signer: NSString, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
+        do {
+            guard let keyStore = geth_node.getKeyStore() else {
+                throw RuntimeError("KeyStore not ready")
+            }
+            let data = Data(hexString: data as String)
+            let signerAccount = try geth_node.findAccount(rawAddress: signer as String)!
+            let signerAddress = signerAccount.getAddress()!
+            let signature = try keyStore.signHash(signerAddress, hash: data)
```

Comment:
is data a hash or raw hex?
docstring is unclear

Reply:
or does your `Data` type take care of this? 🤔

Reply:
The `data` variable is reassigned at line `201` and it transitions from a hex string (NSString) to a byte array (Data). And by the way just to be clear the `Data` type is the swift type of a byte array, what I did above was just _extend_ it with the ability to be initialised from a hex string and to be able to serialise back to a hex string.